pkg/cmd/pulumi/new.go

```diff
@@ -82,6 +82,11 @@ func runNew(ctx context.Context, args newArgs) error {
 		IsInteractive: args.interactive,
 	}
 
+	// Check project name and stack reference project name are the same.
```

Comment:
I wonder if we should just make --stack have to be a plain name. I.e. validate it's a `tokens.Name` not a `QName`. I realize the scuppers the idea of using a fully qualified ref here to set the org, but I think I'm coming around to preferring your idea of adding an "--org" flag to override that. Hell maybe we could even add a prompt as part of the new workflow to give users the choice for org if they haven't set that flag (The backend can suggest which orgs the user has access to, and if they only have one we can just default select it and not prompt so most users wouldn't see any difference here).

This might also interact with your other PR (https://github.com/pulumi/pulumi/pull/13234) so if we like this approach I'd put that PR on a short hold to make this change first.

Reply:
I think this is sensible.

Reply:
This generally sounds reasonable to me, but we should discuss a concrete proposal in ideation.

Though, I don't think we can get away with forcing `--stack` to be only a plain name, because it can be a fully qualified name already today. As such, this change to do this validation looks reasonable to me.

Reply:
>Though, I don't think we can get away with forcing --stack to be only a plain name, because it can be a fully qualified name already today.

I think we've only run into problems trying to pass a ref here internally, I haven't seen any users doing this? I suspect it would be OK to do this.

> As such, this change to do this validation looks reasonable to me.

If we are going to allow a ref here then yes we should validate the project matches but we should do that by calling the backends ParseStackReference method and checking the refs Project() property. Not by doing the string split currently proposed in this change.
That Parse will also cover a lot of other validation of the ref like checking its actually a valid name.